Purchases/Networking/Backend.swift

```diff
@@ -94,691 +79,90 @@ class Backend {
-        return callbackQueue.sync { [self] in
-            var callbacksForKey = createAliasCallbacksCache[key] ?? []
-            let cacheStatus: CallbackCacheStatus = !callbacksForKey.isEmpty
-                ? .addedToExistingInFlightList
-                : .firstCallbackAddedToList
-
-            callbacksForKey.append(createAliasCallback)
-            createAliasCallbacksCache[key] = callbacksForKey
-            return cacheStatus
-        }
-    }
-
-    func add(identifyCallback: @escaping IdentifyResponseHandler, key: String) -> CallbackCacheStatus {
-        return callbackQueue.sync { [self] in
-            var callbacksForKey = identifyCallbacksCache[key] ?? []
-            let cacheStatus: CallbackCacheStatus = !callbacksForKey.isEmpty
-                ? .addedToExistingInFlightList
-                : .firstCallbackAddedToList
-
-            callbacksForKey.append(identifyCallback)
-            identifyCallbacksCache[key] = callbacksForKey
-            return cacheStatus
-        }
-    }
-
-    func getOfferingsCallbacksAndClearCache(forKey key: String) -> [OfferingsResponseHandler] {
-        return callbackQueue.sync { [self] in
-            let callbacks = offeringsCallbacksCache.removeValue(forKey: key)
-            assert(callbacks != nil)
-            return callbacks ?? []
-        }
-    }
-
-    func getCustomerInfoCallbacksAndClearCache(forKey key: String) -> [BackendCustomerInfoResponseHandler] {
-        return callbackQueue.sync { [self] in
-            let callbacks = customerInfoCallbacksCache.removeValue(forKey: key)
-            assert(callbacks != nil)
-            return callbacks ?? []
-        }
-    }
-
-    func getCreateAliasCallbacksAndClearCache(forKey key: String) -> [PostRequestResponseHandler?] {
-        return callbackQueue.sync { [self] in
-            let callbacks = createAliasCallbacksCache.removeValue(forKey: key)
-            assert(callbacks != nil)
-            return callbacks ?? []
-        }
-    }
-
-    func getIdentifyCallbacksAndClearCache(forKey key: String) -> [IdentifyResponseHandler] {
-        return callbackQueue.sync { [self] in
-            let callbacks = identifyCallbacksCache.removeValue(forKey: key)
-            assert(callbacks != nil)
-            return callbacks ?? []
+        let getIntroEligibilityOperation = GetIntroEligibilityOperation(httpClient: self.httpClient,
+                                                                        authHeaders: self.authHeaders)
+        self.operationQueue.addOperation {
+            getIntroEligibilityOperation.getIntroEligibility(appUserID: appUserID,
+                                                             receiptData: receiptData,
+                                                             productIdentifiers: productIdentifiers,
+                                                             completion: completion)
         }
     }
 
 }
-
-// swiftlint:enable type_body_length file_length
```

Comment:
oohhh yesss look at that file size
<img width="112" alt="image" src="https://user-images.githubusercontent.com/3922667/142920491-34346dab-62c7-4f58-b022-411d1f846871.png">

Reply:
😄